### PR TITLE
Feature/processing data

### PR DESCRIPTION
**Pull request by mistermaxis**
### Tasks Completed:
- Add relations to the models
- Add methods to access data

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Comment < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
   belongs_to :post
+
```
Comment (MahmoudBakr23) [inline app/models/comment.rb:4]: **OPTIONAL**
- It'd be nice if you created a custom method using the `after_save` rails callback to call the `increase_comment_counter` method every time the model gets updated!

Ex:
`after_save :increase_comment_counter`

> Please consider this suggestion in the other models

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'Post', foreign_key: 'user_id'
```
Comment (MahmoudBakr23) [inline app/models/user.rb:2]: - [ ] Kindly specify the `foreign_key` for the User model's associations here for the application to know that this specific model is what you're calling the `author`.
Ex:
`has_many :posts, foreign_key: 'author_id'`

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @mistermaxis 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linters are correct
- Descriptive README file and PR
- Well done setting up your models so far

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Please check the comments under the review._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (mistermaxis) [reply app/models/user.rb:2]: Thanks for the suggestion. The foreign key is called user_id, would that still work?

Comment (Deepakdanger) [reply app/models/user.rb:2]: - [ ] As from the previous reviewer and microverse requirements, the foreign key column name should be `author_id`. Please update this.
Hint:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'

```

File: app/models/post.rb [hunk 3]
```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
```
Comment (Deepakdanger) [inline app/models/post.rb:2]: - [ ] In this Please you don't need to mention `foreign_key` explicitly.
Ex:
```ruby
belongs_to :author, class_name: 'User'
```
Change in the `comment.rb` and `like.rb` file as well.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @mistermaxis, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @mistermaxis,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**